### PR TITLE
CE-2013

### PR DESCRIPTION
**Pull request by marutinandanpandya**

### CODE REVIEW

File: UCM-Powershell-IIS/Instalation.ps1 [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+  Write-Host "Servername ?[Default = Localhost]"
+  PrintHintText $script:InstanceNameHintText
+  $InstanceName = Read-Host
+  if ($InstanceName -eq "")
+  {
+    $InstanceName = "localhost"
+  }
+
+#  $AuthArray = "No Authentication", "Windows Authentication"
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/Instalation.ps1]: remove commented code

File: UCM-Powershell-IIS/UCM-Monitor.ps1 [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+[System.Xml.XmlElement]$Metrics = $Config.Settings.MetricGroups
+
+# Setting execution policy to unrestricted so that no user is required to reply to security prompts once script starts running
+Set-ExecutionPolicy Unrestricted
+
+Write-Log "Starting monitoring agent."
+
+
+$counterrrrr = 0
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/UCM-Monitor.ps1]: varname

File: UCM-Powershell-IIS/UCM-Monitor.ps1 [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  [string]$ServiceName = $metric.ServiceName
+  [string]$MetricGroupName = $metric.MetricGroupName
+  [string]$MetricGroupLabel = $metric.MetricGroupLabel
+  [string]$MonitoringFrequency = $metric.Frequency
+  [string]$DashboardName = $metric.DashboardName
+  [System.Xml.XmlElement]$Servers = $metric.Servers
+
+  Create-MetricGroup $ApiServer $Apikey $ServiceName $MetricGroupName $MetricGroupLabel $MonitoringFrequency
+  ForEach ($server in $Servers.ChildNodes)
+  {
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/UCM-Monitor.ps1:42]: indentation

File: UCM-Powershell-IIS/UCM-Monitor.ps1 [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  if($counterrrrr -eq 0) {
+  $job = Start-Job -filepath "$root\worker.ps1" -ArgumentList $arguments
+  Write-Log "Spawning worker thread : $($job|out-string)"
+  $counterrrrr  = 1
+  }
+  }
+}
+
+$SleepTime = 0
+while($TRUE) {
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/UCM-Monitor.ps1]: keep coding practice consistent. Either start braces from same line or next line.

File: UCM-Powershell-IIS/dashboard.json [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+          "type": "timeline",
+          "style": "values",
+          "match": "all",
+          "metric": {
+            "metric_group_name": [
+               [7, "Maximum Connections"]
+            ]
+          }
+        },
+        "98": {
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/dashboard.json]: should be 8

File: UCM-Powershell-IIS/Instalation.ps1 [hunk 6]
```diff
@@ -0,0 +1,336 @@
+
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/Instalation.ps1]: rename file and change calls accordingly

File: UCM-Powershell-IIS/worker.ps1 [hunk 7]
```diff
@@ -0,0 +1,208 @@
+<# This is a worker thread process called by UCM-SQL-Monitor.ps1. Below is a declaration
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/worker.ps1]: update name to Worker.ps1 (all filenames are capital) and update calls accordingly.

Comment (gagan93jtg) [inline UCM-Powershell-IIS/worker.ps1]: Also update comment (UCM-SQL-Monitor thing)

File: UCM-Powershell-IIS/worker.ps1 [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+function Get-PerformanceMetrics
+{
+  Try
+  {
+    if($env:computername -eq $Hostname) {
+      $samples = Get-Counter -Counter $QueryMetrics
+    } else {
+      $samples = Get-Counter -CounterName $Hostname -Counter $QueryMetrics
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/worker.ps1]: CounterName => ComputerName

File: UCM-Powershell-IIS/Instalation.ps1 [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+$SelectedServices = [System.Collections.ArrayList]@()
+
+# Hint texts for user, for the sake of verbosity
+$SampleRateHintText        = "Hint : Monitoring Frequency of 15 seconds means that a sample is sent to us after 15 seconds."
+$MetricGroupNameHintText   = "Hint : This is the name which appears in 'Custom Tab -> Metric Groups -> Name field' in Uptime Cloud Monitor UI."
+$MetricGroupLabelHintText  = "Hint : This is the label which appears in 'Custom Tab -> Metric Groups -> Label field' in Uptime Cloud Monitor UI."
+$DashboardNameHintText     = "Hint : This is the name of dashboard which appears in 'Dashboards' tab and will show all your Monitored Instances at one place."
+$UniqueNameHintText        = "Hint : Unique name represents this particular instance and is visible at Custom -> Custom Objects -> Identifier."
+
+$HostnameHintText          = "Hint : Hostname is the name of the server instance where the SQL server is running.`
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/Instalation.ps1]: Update strings to exclude SQL and add IIS

File: UCM-Powershell-IIS/Instalation.ps1 [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+$UsernameHintText          = "Hint : Name of user which can access system views."
+$PasswordHintText          = "Hint : Password for the user."
+
+
+$LogFile = "$PSScriptRoot\ucm-metrics.log"
+function Write-Log($Message)
+{
+  $TimeStamp =  Get-Date -Format "yyyy-MM-dd hh:mm:ss"
+  $ProcessId = $([System.Diagnostics.Process]::GetCurrentProcess().Id)
+  Add-Content $LogFile -value "$TimeStamp SQL-Powershell-Agent pid:$ProcessId> $Message"
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/Instalation.ps1]: IIS Agent !!!

File: UCM-Powershell-IIS/Instalation.ps1 [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+
+  Write-Host "Servername/Hostname ?"
+  $InstanceName = Read-Host
+  if ($InstanceName -eq "")
+  {
+    Write-Host "Continuing with blank instance name. We will set if to localhost"
+  }
+}
+
+function ConfigureSQLServerAuthentication
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/Instalation.ps1]: rename methods. Also remove username and password in IIS if not required. This goes for code, config file, installation script and everything else.

File: UCM-Powershell-IIS/Utils.ps1 [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+# LogFile variable, used in Write-log function to write all the log content to log file
+$LogFile = "$PSScriptRoot\ucm-metrics.log"
+
+
+function Write-Log($Message)
+{
+  $TimeStamp =  Get-Date -Format "yyyy-MM-dd hh:mm:ss"
+  $ProcessId = $([System.Diagnostics.Process]::GetCurrentProcess().Id)
+  Add-Content $LogFile -value "$TimeStamp SQL-Powershell-Agent pid:$ProcessId> $Message"
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/Utils.ps1]: IIS Agent !!

File: UCM-Powershell-IIS/worker.ps1 [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+
+function Initialize-VariablesFromConfig($Config)
+{
+  $script:Instance         = [string]$Config.InstanceName.trim()
+  $script:SystemIdentifier = [string]$Config.SystemIdentifier.trim()
+  $script:Username         = [string]$Config.Username.trim()
+  $script:Password         = [string]$Config.Password.trim()
+  $script:Hostname         = [string]$Config.Hostname.trim()
+
+  Write-Log "Parsed Argument from config ==> SQL Server Instance name : '$Instance'"
```
Comment (gagan93jtg) [inline UCM-Powershell-IIS/worker.ps1]: IIS everywhere

File: UCM-Powershell-IIS/UCM-Monitor.ps1 (hunk 3 above)
Comment (marutinandanpandya) [reply UCM-Powershell-IIS/UCM-Monitor.ps1:42]: fixed

**Pull request closed without merging**